internal/api/commitment.go

```diff
@@ -1045,3 +1045,100 @@ func (p *v1Provider) getCommitmentConversionRate(source, target core.ResourceBeh
 	toAmount = source.CommitmentConversion.Weight / divisor
 	return fromAmount, toAmount
 }
+
+// ExtendCommitmentDuration handles POST /v1/domains/{domain_id}/projects/{project_id}/commitments/{commitment_id}/extend-duration
+func (p *v1Provider) UpdateCommitmentDuration(w http.ResponseWriter, r *http.Request) {
+	httpapi.IdentifyEndpoint(r, "/v1/domains/:domain_id/projects/:project_id/commitments/:commitment_id/extend-duration")
+	token := p.CheckToken(r)
+	if !token.Require(w, "project:edit") {
+		return
+	}
+	commitmentID := mux.Vars(r)["commitment_id"]
+	if commitmentID == "" {
+		http.Error(w, "no transfer token provided", http.StatusBadRequest)
+		return
+	}
+	dbDomain := p.FindDomainFromRequest(w, r)
+	if dbDomain == nil {
+		return
+	}
+	dbProject := p.FindProjectFromRequest(w, r, dbDomain)
+	if dbProject == nil {
+		return
+	}
+	var Request struct {
+		Duration limesresources.CommitmentDuration `json:"duration"`
+	}
+	req := Request
+	if !RequireJSON(w, r, &req) {
+		return
+	}
+
+	var dbCommitment db.ProjectCommitment
+	err := p.DB.SelectOne(&dbCommitment, findProjectCommitmentByIDQuery, commitmentID, dbProject.ID)
+	if errors.Is(err, sql.ErrNoRows) {
+		http.Error(w, "no such commitment", http.StatusNotFound)
+		return
+	} else if respondwith.ErrorText(w, err) {
+		return
+	}
+	var loc datamodel.AZResourceLocation
+	err = p.DB.QueryRow(findProjectAZResourceLocationByIDQuery, dbCommitment.AZResourceID).
+		Scan(&loc.ServiceType, &loc.ResourceName, &loc.AvailabilityZone)
+	if errors.Is(err, sql.ErrNoRows) {
+		// defense in depth: this should not happen because all the relevant tables are connected by FK constraints
+		http.Error(w, "no route to this commitment", http.StatusNotFound)
+		return
+	} else if respondwith.ErrorText(w, err) {
+		return
+	}
+	behavior := p.Cluster.BehaviorForResource(loc.ServiceType, loc.ResourceName)
+	validDurations := behavior.CommitmentDurations
+
+	existsInDurations := false
+	for _, duration := range validDurations {
+		if duration.String() == req.Duration.String() {
+			existsInDurations = true
+		}
+	}
+	if !existsInDurations {
```

Comment:
There is no relevant ambiguity in how commitment durations are parsed and serialized, so we can use the same check as parseAndValidateCommitmentRequest here:

```suggestion
	if !slices.Contains(behavior.CommitmentDurations, req.Duration) {
```